code/__HELPERS/unsorted.dm

```diff
@@ -504,21 +504,19 @@ Turf and target are separate in case you want to teleport some distance from a t
 	return y
 
 /*
-Recursively gets all contents of contents and returns them all in a list.
-
-recursive_depth is useful if you only want a turf and everything on it (recursive_depth=1)
-Do not set recursive depth higher than MAX_PROC_DEPTH as byond breaks when that limit is reached.
+	Gets all contents of contents and returns them all in a list.
 */
-/atom/proc/GetAllContents(list/output=list(), recursive_depth=MAX_PROC_DEPTH, _current_depth=0)
-	. = output
-	output += src 
-	if(_current_depth == recursive_depth)
-		if(_current_depth == MAX_PROC_DEPTH)
-			WARNING("Get all contents reached the max recursive depth of [MAX_PROC_DEPTH]. More and we would break shit. Offending atom: [src]")
-		return
-	for(var/i in 1 to contents.len) 
-		var/atom/thing = contents[i] 
-		thing.GetAllContents(output, recursive_depth, ++_current_depth) 
+/atom/proc/GetAllContents()
+	var/list/processing_list = list(src)
+	var/list/assembled = list()
+	while(processing_list.len)
+		var/atom/A = processing_list[1]
+		processing_list -= A
```

Comment:
Surely `.Cut(1, 2)` would be faster, no?

Reply:
depends how it's implemented.
I'll check again in like, 10 mins.

Reply:
![image](https://user-images.githubusercontent.com/4043781/31893911-bed37d9a-b804-11e7-9c98-b7ba7997a41d.png)

@Cyberboss minuscule difference over 1 mil calls.

Reply:
not worthwhile enough to hold up the pr over imo but can be done later if you really want

Reply:
@duncathan if he really wants to, maintainer edits are enabled on the PR already.